### PR TITLE
Spec: First stab at specifying Grammar

### PR DESCRIPTION
**Pull request by Yoric**

### CODE REVIEW

File: spec/context.md [hunk 1]
```diff
@@ -13,7 +13,66 @@ This document specifies how to compress an Abstract Syntax Tree (or AST) in a gi
...
+
+```typescript
+type Grammar = Set<Declaration>;
+
+class Declaration {
+  name: String;
+  content: Content;
+};
+class Content {
+  optional: bool;
```
Comment (arai-a) [inline spec/context.md]: what is `optional` field for?

Comment (Yoric) [reply spec/context.md]: For optional values. I should document this.

File: spec/context.md [hunk 2]
```diff
@@ -13,7 +13,66 @@ This document specifies how to compress an Abstract Syntax Tree (or AST) in a gi
...
+           | Interface;
+
+class PrimitiveString { }
+class PrimitiveF64 { }
+class PrimitiveU32 { }
+class PrimitiveBool { }
+class PrimitivePropertyKey { }
+class PrimitiveIdentifierName { }
+class Array {
+  values: Definition
```
Comment (arai-a) [inline spec/context.md]: shouldn't this be `Content` (if nullable) or `Value` (non-nullable) ?
I don't think it elements should be named.

**Comment by Yoric** [comment]
Closing PR. We might resurrect this at some point after #428 has landed.

**Pull request closed without merging**